src/test/java/uk/gov/companieshouse/company/metrics/service/MetricsApiResponseHandlerTest.java

```diff
@@ -0,0 +1,136 @@
+    private static final String CONTEXT_ID = "1234";
+
+    @Mock
+    private Logger logger;
+
+    @Mock
+    private IllegalArgumentException illegalArgumentException;
+
+    @Mock
+    private Throwable throwable;
+
+    @InjectMocks
+    private MetricsApiResponseHandlerImpl responseHandler;
+
+    @Test
+    void testHandleUriValidationException() {
+        // given
+        String message = String.format(FAILED_MSG, APPOINTMENTS_DELTA_TYPE, COMPANY_NUMBER, CONTEXT_ID);
+
+        // when
+        Executable actual = () -> responseHandler.handle(COMPANY_NUMBER, APPOINTMENTS_DELTA_TYPE, new URIValidationException(any()), CONTEXT_ID);
+
+        // then
+        Exception ex = assertThrows(NonRetryableErrorException.class, actual);
+        assertEquals(message, ex.getMessage());
+        verify(logger).error(message);
+    }
+
+    @Test
+    void testHandleIllegalArgumentException() {
+        // given
+        String message = String.format(FAILED_MSG, APPOINTMENTS_DELTA_TYPE, COMPANY_NUMBER, CONTEXT_ID);
+        String causeMessage = "cause message";
+        when(illegalArgumentException.getCause()).thenReturn(throwable);
+        when(throwable.getMessage()).thenReturn(causeMessage);
+
+        // when
+        Executable actual = () -> responseHandler.handle(COMPANY_NUMBER, APPOINTMENTS_DELTA_TYPE, illegalArgumentException, CONTEXT_ID);
+
+        // then
+        Exception ex = assertThrows(RetryableErrorException.class, actual);
+        assertEquals(message, ex.getMessage());
+        verify(logger).info(message + "; " + causeMessage);
+    }
+
+    @Test
+    void testHandleIllegalArgumentExceptionWhenCauseIsNull() {
+        // given
+        String message = String.format(FAILED_MSG, APPOINTMENTS_DELTA_TYPE, COMPANY_NUMBER, CONTEXT_ID);
+
+        // when
+        Executable actual = () -> responseHandler.handle(COMPANY_NUMBER, APPOINTMENTS_DELTA_TYPE, illegalArgumentException, CONTEXT_ID);
+
+        // then
+        Exception ex = assertThrows(RetryableErrorException.class, actual);
+        assertEquals(message, ex.getMessage());
+        verify(logger).info(message);
+    }
+
+    @Test
+    void testHandleApiErrorResponseExceptionWhenStatusCode500() {
+        // given
+        int statusCode = 500;
+        String message = String.format(ERROR_MSG, statusCode, APPOINTMENTS_DELTA_TYPE, COMPANY_NUMBER, CONTEXT_ID);
+        HttpResponseException.Builder builder = new HttpResponseException.Builder(statusCode, "", new HttpHeaders());
+
+        // when
+        Executable actual = () -> responseHandler.handle(COMPANY_NUMBER, APPOINTMENTS_DELTA_TYPE, new ApiErrorResponseException(builder), CONTEXT_ID);
+
+        // then
+        Exception ex = assertThrows(RetryableErrorException.class, actual);
+        assertEquals(message, ex.getMessage());
+        verify(logger).info(message);
+    }
+
+    @Test
+    void testHandleApiErrorResponseExceptionWhenStatusCode404() {
+        // given
+        int statusCode = 404;
```

Comment:
Similar here, HttpStatus values could be used in below String.format and HttpResponseException Builder instead